lib/matplotlib/widgets.py

```diff
@@ -1670,13 +1835,13 @@ def __init__(self, ax, labels, active=0, activecolor=None, *,
         radio_props.setdefault('edgecolor', radio_props.get('color', 'black'))
         radio_props.setdefault('facecolor',
                                radio_props.pop('color', activecolor))
-        self._buttons = ax.scatter([.15] * len(ys), ys, **radio_props)
+        self._buttons = ax.scatter(button_xs, button_ys, **radio_props)
         # The user may have passed custom colours in radio_props, so we need to
         # create the radios, and modify the visibility after getting whatever
         # the user set.
         self._active_colors = self._buttons.get_facecolor()
         if len(self._active_colors) == 1:
-            self._active_colors = np.repeat(self._active_colors, len(labels),
+            self._active_colors = np.repeat(self._active_colors, n_labels,
```

Comment:
```suggestion
            self._active_colors = np.repeat(self._active_colors, len(labels),
```
Revert, as there is no benefit in declaring an additional variable.